test/PushCore/unit_tests/ChannelCreationPush/ChannelMigration.t.sol

```diff
@@ -0,0 +1,138 @@
+pragma solidity ^0.8.20;
+
+import {BasePushCoreTest} from "../BasePushCoreTest.t.sol";
+import {CoreTypes} from "../../../../contracts/libraries/DataTypes.sol";
+import {EPNSCoreProxy, ITransparentUpgradeableProxy} from "contracts/PushCore/EPNSCoreProxy.sol";
+import { PushCoreMock } from "contracts/mocks/PushCoreMock.sol";
+import {console} from "forge-std/console.sol";
+import { PushCoreV3 } from "contracts/PushCore/PushCoreV3.sol";
+import { EPNSCoreProxy, ITransparentUpgradeableProxy } from "contracts/PushCore/EPNSCoreProxy.sol";
+
+contract Migration_Test is BasePushCoreTest {
+        PushCoreMock public coreV2;
+        PushCoreV3 public coreV3;
+        EPNSCoreProxy public epnsCoreProxyV2;
+
+    function setUp() public virtual override {
+        BasePushCoreTest.setUp();
+        coreV2 = new PushCoreMock();
+        coreV3 = new PushCoreV3();
+        epnsCoreProxyV2 = new EPNSCoreProxy(
+            address(coreV2),
+            address(epnsCoreProxyAdmin),
+            actor.admin,
+            address(pushToken),
+            address(0), // WETH Address
+            address(uniV2Router), // Uniswap_Router
+            address(0), // Lending_Pool_Aave
+            address(0), // Dai_Address
+            address(0), // aDai address
+            0
+        );
+        coreV2 = PushCoreMock(address(epnsCoreProxyV2));
+        changePrank(tokenDistributor);
+        pushToken.transfer(address(coreV2 ), 1 ether);
+
+        // Approve tokens of actors now to core contract proxy address
+        approveTokens(actor.admin, address(coreV2), 50_000 ether);
+        approveTokens(actor.governance, address(coreV2), 50_000 ether);
+        approveTokens(actor.bob_channel_owner, address(coreV2), 50_000 ether);
+        approveTokens(actor.alice_channel_owner, address(coreV2), 50_000 ether);
+        approveTokens(actor.charlie_channel_owner, address(coreV2), 50_000 ether);
+        approveTokens(actor.dan_push_holder, address(coreV2), 50_000 ether);
+        approveTokens(actor.tim_push_holder, address(coreV2), 50_000 ether);  
+        
+    }
+
+    function test_ProtocolPoolFeesCorrectForMultipleChannelsCreation()
```

Comment:
The name could be improved here as well: test_ProtocolPoolFees_IsCorrect_ForMultipleChannelsCreation